plugins/github/test/samson_github/samson_plugin_test.rb

```diff
@@ -58,4 +58,38 @@
       Samson::Hooks.fire(:before_deploy, deploy, nil)
     end
   end
+
+  describe :repo_provider_status do
+    def fire
+      Samson::Hooks.fire(:repo_provider_status).first
+    end
+
+    let(:status_url) { "#{SamsonGithub::STATUS_URL}/api/status.json" }
+
+    around { |t| Samson::Hooks.only_callbacks_for_plugin('github', :repo_provider_status, &t) }
+
+    it "reports good response" do
+      assert_request(:get, status_url, to_return: {body: {status: 'good'}.to_json}) do
+        refute fire
```

Comment:
Might be good to be a little more specific, and assert on the actual object we expect to get back, such as `nil` or the expected array of errors?